configuration_manager/xml_parser.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2020 Forschungszentrum Jülich GmbH and Aix-Marseille Université
+# "Licensed to the Apache Software Foundation (ASF) under one or more
+#  contributor license agreements; and to You under the Apache License,
+#  Version 2.0."
+
+import xml.etree.ElementTree as ET
+
+
+class Parser(object):
+    """
+    This class provides functions for parsing an xml file
+    and converting it into Python dictionary type for manipulation.
+    """
+
+    @classmethod
+    def load_xml(cls, file):
+        """Parses an XML section into an element tree.
+
+        Parameters
+        ----------
+        file : object
+            XML file containg XML data
+
+        Returns
+        -------
+        xmltree: ElementTree instance
+            The parsed XML
+        """
+        try:
+            cls.__xmltree = ET.parse(file)
+        except FileNotFoundError as e:
+            raise e  # TODO: a better exception handling
+        else:
+            return cls.__xmltree
+
+    @classmethod
+    def __build_child_nodes(cls, child_list):
+        """Helper function to build nested items.
+
+        Parameters
+        ----------
+        child_list :
+            The parent element of the nested elements.
```

Comment:
Function name does not match the description.
The name of the parameters does not match the type (list vs xml tree)
It is not clear from the description what this function does